aospy/test/test_data_loader.py

```diff
@@ -0,0 +1,392 @@
+                          coords=[time, lat, lon],
+                          dims=[TIME_STR, self.ALT_LAT_STR, LON_STR],
+                          name=self.var_name).to_dataset()
+        ds[TIME_BOUNDS_STR] = xr.DataArray(time_bounds,
+                                           coords=[time, nv],
+                                           dims=[TIME_STR, NV_STR],
+                                           name=TIME_BOUNDS_STR)
+        units_str = 'days since 2000-01-01 00:00:00'
+        ds[TIME_STR].attrs['units'] = units_str
+        ds[TIME_BOUNDS_STR].attrs['units'] = units_str
+        self.ds = ds
+
+        inst_time = np.array([3, 6, 9])
+        inst_units_str = 'hours since 2000-01-01 00:00:00'
+        inst_ds = ds.copy()
+        inst_ds.drop(TIME_BOUNDS_STR)
+        inst_ds[TIME_STR].values = inst_time
+        inst_ds[TIME_STR].attrs['units'] = inst_units_str
+        self.inst_ds = inst_ds
+
+    def tearDown(self):
+        pass
+
+
+class TestDataLoader(AospyDataLoaderTestCase):
+    def test_rename_grid_attrs(self):
+        assert LAT_STR not in self.ds
+        assert self.ALT_LAT_STR in self.ds
+        ds = self.DataLoader.rename_grid_attrs(self.ds)
+        assert LAT_STR in ds
+
+    def test_set_grid_attrs_as_coords(self):
+        ds = self.DataLoader.rename_grid_attrs(self.ds)
+        sfc_area = ds[self.var_name].isel(**{TIME_STR: 0}).drop(TIME_STR)
+        ds[SFC_AREA_STR] = sfc_area
+
+        # Assert that SFC_AREA_STR is not initially carried by self.var_name
+        assert SFC_AREA_STR not in ds[self.var_name]
+
+        # Apply method, then assert that SFC_AREA_STR is carried
+        ds = self.DataLoader.set_grid_attrs_as_coords(ds)
+        assert SFC_AREA_STR in ds[self.var_name]
+
+    def test_sel_var(self):
+        time = np.array([0, 31, 59]) + 15
+        data = np.zeros((3))
+        ds = xr.DataArray(data,
+                          coords=[time],
+                          dims=[TIME_STR],
+                          name=convection_rain.name).to_dataset()
+        condensation_rain_alt_name, = condensation_rain.alt_names
+        ds[condensation_rain_alt_name] = xr.DataArray(data, coords=[ds.time])
+        result = self.DataLoader._sel_var(ds, convection_rain)
+        self.assertEqual(result.name, convection_rain.name)
+
+        result = self.DataLoader._sel_var(ds, condensation_rain)
+        self.assertEqual(result.name, condensation_rain.name)
+
+        with self.assertRaises(KeyError):
+            self.DataLoader._sel_var(ds, precip)
+
+    def test_maybe_apply_time_shift(self):
+        ds = xr.decode_cf(self.ds)
+        da = ds[self.var_name]
+
+        # If you don't pass a time_offset or non-null DataVars,
+        # nothing should happen
+        result = self.DataLoader._maybe_apply_time_shift(da.copy())[TIME_STR]
+        assert result.identical(da[TIME_STR])
+
+        # If you do pass an offset, it should be applied
+        # Why does da get modified in place?
```

Comment:
Yes, I am wondering why ```_maybe_apply_time_shift``` modifies its DataArray argument in place.  Ideally I wouldn't want it to.